### PR TITLE
feat(qr): Added QR scanner

### PR DESCRIPTION
**Pull request by yehjxraymond**

### CODE REVIEW

File: src/services/qrProcessor/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { get } from "axios";
+
+export const decodeQrCode = qrCode => {
+  const ttRegex = /tt:\/\/(.*)/;
```
Comment (rjchow) [inline src/services/qrProcessor/index.js]: tt:// is too generic there's probably some other app in the wild using it
go with tradetrust://
also namespace the action because if there's a mobile app in the future we'll probably use the deeplinking feature for more things than this
so
tradetrust://view-document?target=\<encoded url\>

refer to
https://portal.deeplink.me/documentation 
under schemes and url handling

File: src/components/CertificateDropZone/CertificateDropZoneContainer.js [hunk 2]
```diff
@@ -12,27 +12,38 @@ import {
...
+  getStoreStatus,
+  processQrCode
 } from "../../reducers/certificate";
 import { updateNetworkId } from "../../reducers/application";
 import CertificateDropZone from "./CertificateDropZone";
+import css from "./Views/viewerStyles.scss";
+import QrReader from "../QrReader";
 
-class CertificateDropZoneContainer extends Component {
+export class CertificateDropZoneContainer extends Component {
```
Comment (sumitnitsurat) [inline src/components/CertificateDropZone/CertificateDropZoneContainer.js:23]: Not needed this export. Already exporting at the end.

File: src/components/CertificateDropZone/CertificateDropZoneContainer.js [hunk 3]
```diff
@@ -42,26 +53,44 @@ class CertificateDropZoneContainer extends Component {
...
+        <button
+          type="button"
+          onClick={this.toggleQrReaderVisible}
+          className={`pointer ${css.btn} ${css["disable-btn"]}`}
+        >
+          Disable
+        </button>
+      </>
+    ) : (
+      <>
```
Comment (sumitnitsurat) [inline src/components/CertificateDropZone/CertificateDropZoneContainer.js]: not needed fragment also. Only a single child is there.

File: src/components/CertificateDropZone/CertificateDropZoneContainer.js [hunk 4]
```diff
@@ -82,14 +111,17 @@ const mapStateToProps = store => ({
 const mapDispatchToProps = dispatch => ({
   updateNetworkId: () => dispatch(updateNetworkId()),
   updateCertificate: payload => dispatch(updateCertificate(payload)),
-  resetData: () => dispatch(resetCertificateState())
+  resetData: () => dispatch(resetCertificateState()),
+  processQr: payload => dispatch(processQrCode(payload))
 });
 
-export default connect(
+const ConnectedCertificateDropZoneContainer = connect(
```
Comment (sumitnitsurat) [inline src/components/CertificateDropZone/CertificateDropZoneContainer.js:116]: why this change ? its the same just a line extra.

File: src/components/QrReader/qrReader.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+const QrReaderZone = ({ handleQrScanned }) => {
+  const onScan = data => {
+    if (data) handleQrScanned(data);
+  };
+
+  return (
+    <QrReader
+      data-id="qr-code-reader"
+      delay={100}
+      onError={error}
```
Comment (sumitnitsurat) [inline src/components/QrReader/qrReader.js:20]: No handling of the error ? log the error with text also.

File: src/services/qrProcessor/index.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { get } from "axios";
```
Comment (sumitnitsurat) [inline src/services/qrProcessor/index.js:1]: cant we use fetch instead ?

File: src/components/CertificateDropZone/CertificateDropZoneContainer.js (hunk 2 above)
Comment (yehjxraymond) [reply src/components/CertificateDropZone/CertificateDropZoneContainer.js:23]: Exporting the version without `connect()` for the unit test on the container alone.

File: src/components/CertificateDropZone/CertificateDropZoneContainer.js (hunk 4 above)
Comment (yehjxraymond) [reply src/components/CertificateDropZone/CertificateDropZoneContainer.js:116]: Same reason as above.
- Default export the connected version
- Named export for the non-connected version for unit test (we are not testing the connect feature and it messes around with the test)

File: src/components/QrReader/qrReader.js (hunk 5 above)
Comment (yehjxraymond) [reply src/components/QrReader/qrReader.js:20]: The error handling uses the logger under the namespace `services:qr` already. See L9

File: src/services/qrProcessor/index.js (hunk 6 above)
Comment (yehjxraymond) [reply src/services/qrProcessor/index.js:1]: `fetch` requires the isomorphic fetch import. Perfer having axios over fetch since its api is consistent in node and browser environment and testing is more straightforward. 

You just need `jest.mock("axios");`

File: src/components/CertificateDropZone/CertificateDropZoneContainer.js (hunk 3 above)
Comment (yehjxraymond) [reply src/components/CertificateDropZone/CertificateDropZoneContainer.js]: Removed!

File: src/services/qrProcessor/index.js (hunk 1 above)
Comment (yehjxraymond) [reply src/services/qrProcessor/index.js]: Changed to tradetrust

File: src/components/QrReader/qrReader.js (hunk 5 above)
Comment (sumitnitsurat) [reply src/components/QrReader/qrReader.js:20]: we can put the what is the error also ? currently we are logging "services:qr" text only. we dont know the error.

Comment (yehjxraymond) [reply src/components/QrReader/qrReader.js:20]: That method logs the ouput, see debug's implm: https://www.npmjs.com/package/debug

Comment (sumitnitsurat) [reply src/components/QrReader/qrReader.js:20]: ahh..okay.
